### ISSUE TITLE
Potentially not working in certain areas after recent client update

### ISSUE THREAD

**Issue by Matt17BR** [abba23/spotify-adblock-linux#108]
Hi, I'm running the AUR build of this, and even though spotify-adblock got updated to r82.d458130-1 along with the latest spotify spotify testing package (1.55.xxx), the ad blocker doesn't seem to be working as expected, or at least it doesn't seem to in Italy, since other commenters on AUR haven't experienced any trouble. I'll just copy paste my initial AUR comment here:

"i've rebuilt spotify-adblock-git and i'm on version r82.d458130-1. however, this new version doesn't seem to play well with the latest spotify stable branch update (1.55). i can listen to music fine, and search works properly, however artist profiles, albums, as well as the home page can't be loaded at all. i'm pretty sure it's not normal behaviour since the dev seems to have updated the app with support for the latest version, so i'd like to hear if anyone has encountered the same issue, thanks."

AUR-maintainer xyzzy reccomended I launch the adblocker via LD_PRELOAD=/usr/lib/spotify-adblock.so spotify to debug address requests a bit, and I think I might have found the culprit, i.e. the address gew-spclient.spotify.com which is apparently used to load artist, album and home pages in certain select areas. I got a segfault at some point while trying to load the home page too. I didn't find that address in the whitelist so I think that could potentially be it? I'm not too sure because I'm a bit inept and I didn't build the package with a different whitelist. Let me know if I'm missing something or could add any idditional info. Attached the terminal log below:
[ld-preload.txt](https://github.com/abba23/spotify-adblock-linux/files/6178541/ld-preload.txt)

**Comment by abba23** [comment]
Thanks for the report. Blocking requests to `gew-spclient.spotify.com` doesn't break anything for me, so I don't think that's the culprit. I can however see blocked requests to `xpui.app.spotify.com` in your log, which don't appear in mine.

Since I can't reproduce this myself, you'll have to test it by adding that domain to the whitelist and building yourself. Just follow the build steps in the readme, but add `xpui.app.spotify.com` to `whitelist.h` before running `make`. Then launch Spotify using `LD_PRELOAD=./spotify-adblock.so spotify` (in the same directory) and see if it works.

**Comment by Matt17BR** [comment]
Right, followed those steps but the issue didn't seem to get fixed, I first tried adding both `xpui.app.spotify.com` and then `gew-spclient.spotify.com` to the whitelist before building as instructed in readme but that didn't do the trick.

However! What fixed it was whitelisting `gew-dealer.spotify.com` ! So that seems to solve the issue. Will try using the client a bit and see if there's any adverse effects, but I doubt so.

**Comment by GuessWhatBBQ** [comment]
Can confirm the testing by Matt17BR partially. Whitelisting both `api-partner.spotify.com` and `api.spotify.com` was required for spotify to be functional, I had the same issues outlined above by Matt17BR.

Now regarding the `gew-dealer.spotify.com`, my guess is that it's specific to Europe West as specified by the gEW at the beginning of the url, and after doing a dns lookup I saw that it has an alias `gew-dealer-ssl.spotify.com` which is most likely an ssl server. Furthermore it doesnt show up on my logs and the closest thing I could find is `dealer.spotify.com` which shows up in both our logs as a whitelisted connection, and after doing a dns lookup on `dealer.spotify.com` it ends up having an alias of `global-dealer-ssl.spotify.com` most likely another ssl server.

On a side note, after messing around with dns lookups in `dig` I ended up finding `gae-dealer.spotify.com` which has an alias `gae-dealer-ssl.spotify.com`. I'm sure you're noticing a pattern here.

All this makes me believe that your client was facing difficulties with the global server and thus tried connecting to one closer to your region and ended up getting blacklisted as it had a different url pattern than the whitelisted one. This could easily be solved by modifying the whitelist to `*dealer.spotify.com`, as there is already an entry for it, to account for any such cases in the future.

Testing from @Matt17BR regarding my solution would be appreciated. Also could you try to whitelist just the api urls to see if the issue with the dealer urls was a one off incident ?

Basically revert any changes to the whitelist.h file then add
```
"api-partner.spotify.com",
"api.spotify.com",
```
And if that doesnt work just modify `"dealer.spotify.com"` to `"*dealer.spotify.com"` at line 37.

**Comment by Matt17BR** [comment]
> Can confirm the testing by Matt17BR partially. Whitelisting both `api-partner.spotify.com` and `api.spotify.com` was required for spotify to be functional, I had the same issues outlined above by Matt17BR.
> 
> Now regarding the `gew-dealer.spotify.com`, my guess is that it's specific to Europe West as specified by the gEW at the beginning of the url, and after doing a dns lookup I saw that it has an alias `gew-dealer-ssl.spotify.com` which is most likely an ssl server. Furthermore it doesnt show up on my logs and the closest thing I could find is `dealer.spotify.com` which shows up in both our logs as a whitelisted connection, and after doing a dns lookup on `dealer.spotify.com` it ends up having an alias of `global-dealer-ssl.spotify.com` most likely another ssl server.
> 
> On a side note, after messing around with dns lookups in `dig` I ended up finding `gae-dealer.spotify.com` which has an alias `gae-dealer-ssl.spotify.com`. I'm sure you're noticing a pattern here.
> 
> All this makes me believe that your client was facing difficulties with the global server and thus tried connecting to one closer to your region and ended up getting blacklisted as it had a different url pattern than the whitelisted one. This could easily be solved by modifying the whitelist to `*dealer.spotify.com`, as there is already an entry for it, to account for any such cases in the future.
> 
> Testing from @Matt17BR regarding my solution would be appreciated. Also could you try to whitelist just the api urls to see if the issue with the dealer urls was a one off incident ?
> 
> Basically revert any changes to the whitelist.h file then add
> ```
> "api-partner.spotify.com",
> "api.spotify.com",
> ```
> And if that doesnt work just modify `"dealer.spotify.com"` to `"*dealer.spotify.com"` at line 37.

When testing yesterday I pretty much tested all combinations of those 3 URLs, in order to check what whitelisting each one did and whether anyone was redundant, so no, unfortunately you need all 3. I'll try testing whether whitelisting *dealer works as soon as I'm home though, that does make sense!

**Comment by GuessWhatBBQ** [comment]
Looks like you've been pretty thorough hehe. Your research definitely saved me from a headache.

**Comment by Matt17BR** [comment]
@GuessWhatBBQ I can now confirm that everything seems to work as expected by replacing `dealer.spotify.com` with `*dealer.spotify.com` on line 37 of the whitelist, as well as adding the two api links mentioned above.

**Comment by GuessWhatBBQ** [comment]
It would be helpful if more people could test the solution before/if it's pushed to master. It's working fine for me now tho, which is better than it was a few hours ago when it was almost unusable.

@abba23 any chance of patching master so we can go back to using the AUR version ?

**Comment by abba23** [comment]
Nice work! I'll just go ahead and add `api.spotify.con`, `api-partner.spotify.com` and `*dealer.spotify.com` to the whitelist.
